fields/types/Type.js

```diff
@@ -64,12 +64,25 @@ function Field(list, path, options) {
 			'Please provide a default, remove the required setting, or set initial: false to override this error.\n');
 		process.exit(1);
 	}
-
+	
+	// if dependsOn and required, set required to a function for validation
+	if (this.options.dependsOn && this.options.required === true) {
+		var opts = this.options;
+		this.options.required = function() {
+			// `this` refers to the validating document
+			debug('validate dependsOn required', evalDependsOn(opts.dependsOn, this.toObject()));
```

Comment:
Without it working in the Admin UI would produce a different result that working on a List directly.